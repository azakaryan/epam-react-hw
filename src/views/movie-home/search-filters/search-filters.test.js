import React from 'react';
import { shallow } from 'enzyme';
import expectEnzymify from 'expect';
import enzymify from 'expect-enzyme';
import SearchFilters from './search-filters';

expectEnzymify.extend(enzymify());

const onFiltersApply = () => {};

describe('Search Filters Component', () => {
  it('should render correctly', () => {
    const component = shallow(<SearchFilters onFiltersApply={onFiltersApply} />);

    expect(component).toMatchSnapshot();
  });

  it('renders a search filters component', () => {
    const wrapper = shallow(<SearchFilters onFiltersApply={onFiltersApply} />);
    expectEnzymify(wrapper).toBeA('div');
  });

  it('set state on search filters component', () => {
    const wrapper = shallow(<SearchFilters onFiltersApply={onFiltersApply} />);
    const titleEl = wrapper.find({ id: 'title' });
    const genreEl = wrapper.find({ id: 'genre' });

    titleEl.simulate('click');
    expect(wrapper.state('searchBy')).toEqual('title');
    titleEl.simulate('click');
    expect(wrapper.state('searchBy')).toEqual('');

    genreEl.simulate('click');
    expect(wrapper.state('searchBy')).toEqual('genres');
    genreEl.simulate('click');
    expect(wrapper.state('searchBy')).toEqual('');
  });

  it('check default state of "search" to be equal to "" on search filters component', () => {
    const wrapper = shallow(<SearchFilters onFiltersApply={onFiltersApply} />);
    expect(wrapper.state('search')).toEqual('');
  });
  it('check default state of "searchBy" to be equal to "" on search filters component', () => {
    const wrapper = shallow(<SearchFilters onFiltersApply={onFiltersApply} />);
    expect(wrapper.state('searchBy')).toEqual('');
  });

  it('simulate filter click on search filters component', () => {
    let k = 1;
    const onFiltersApplyFunc = () => k = k + 1;
    const wrapper = shallow(<SearchFilters onFiltersApply={onFiltersApplyFunc} />);
    expect(k).toEqual(1);

    wrapper.find({ id: 'title' }).simulate('click');
    expect(k).toEqual(1);
    wrapper.find({ id: 'title' }).simulate('click');
    expect(k).toEqual(1);
    wrapper.find({ id: 'genre' }).simulate('click');
    expect(k).toEqual(1);
    wrapper.find({ id: 'apply-search' }).simulate('click');
    expect(k).toEqual(2);
  });

  it('simulate filter input fill on search filters component', () => {
    let k = 1;
    const onFiltersApplyFunc = () => k = k + 1;
    const wrapper = shallow(<SearchFilters onFiltersApply={onFiltersApplyFunc} />);
    wrapper.find({ id: 'search-text' }).simulate('change', { target: { value: 'hello' } });
    expect(k).toEqual(1);
  });
});
