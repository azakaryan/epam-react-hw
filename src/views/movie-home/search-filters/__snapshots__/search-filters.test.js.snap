// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search Filters Component should render correctly 1`] = `
<div
  className="main_container"
>
  <label
    className="label"
  >
    FIND YOUR MOVIE
  </label>
  <div
    className="input_wrapper"
  >
    <input
      id="search-text"
      onChange={[Function]}
      type="text"
      value=""
    />
  </div>
  <div
    className="filters_wrapper"
  >
    <label>
      SEARCH BY
    </label>
    <div
      className="search_by_btns"
    >
      <WithStyles(Button)
        color="default"
        id="title"
        onClick={[Function]}
        variant="contained"
      >
        TITLE
      </WithStyles(Button)>
      <WithStyles(Button)
        color="default"
        id="genre"
        onClick={[Function]}
        variant="contained"
      >
        GENRE
      </WithStyles(Button)>
    </div>
    <div
      className="search_btn"
    >
      <WithStyles(Button)
        color="primary"
        id="apply-search"
        onClick={[Function]}
        variant="contained"
      >
        SEARCH
      </WithStyles(Button)>
    </div>
  </div>
</div>
`;
